app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/ResourcesGridPresenter.kt

```diff
@@ -102,8 +102,8 @@ class ResourcesGridPresenter(
         }
 
         val id = item.id()
-        val metadata = metadataProcessor.retrieve(id).getOrThrow()
-        val preview = previewProcessor.retrieve(id).getOrThrow()
+        val metadata = metadataProcessor.retrieve(id).getOrNull()
```

Comment:
I think we always must have metadata for a resource

Reply:
Better avoid `null` anyway. E.g. when we'll have async metadata generation, something like this API would be desired:
```
fun retrieve(id: ResourceId): Either<Metadata, Job<Metadata>>
```
So, `Left` case would contain final result and `Right` could contain the coroutine.

Reply:
Makes sense